src/algebra/layers.jl

```diff
@@ -74,14 +74,28 @@ function compute_axes_grid(fig, s::OneOrMoreLayers;
     # fit scales (compute plot values using all data values)
     map!(fitscale, values(scales))
 
-    axs = compute_grid_positions(scales)
 
-    axes_grid = map(CartesianIndices(axs)) do c
+    function create_axis(fig, c)
         type = get(axis, :type, Axis)
         options = Base.structdiff(axis, (; type))
         ax = type(fig[Tuple(c)...]; options...)
         return AxisEntries(ax, Entry[], scales)
     end
+    function create_axis(f::Axis, c)
+        if !isempty(axis)
+            @warn("Axis got passed, but also axis attributes. Ignoring axis attributes: $(axis)")
+        end
+        return AxisEntries(fig, Entry[], scales)
+    end
+    axs = compute_grid_positions(scales)
+    sizes = length.(axs)
```

Comment:
```suggestion
    sizes = map(length, axs)
```

I try to consistently use `map` instead of `broadcast` for tuples (not sure if it makes a big difference though).

Reply:
what difference do you expect?

Reply:
It's just an internal style guide to use `map` rather than `broadcast` for simple cases to avoid the compilation overhead of the broadcast machinery. That being said, definitely not a big deal, `broadcast` ends up calling `map` for tuples anyways.